eng/build.sh

```diff
@@ -134,6 +144,22 @@ while [[ $# > 0 ]]; do
       arguments="$arguments /p:LibrariesConfiguration=$2"
       shift 2
       ;;
+     -cross)
+      arguments="$arguments /p:CrossBuild=True"
+      shift 1
+      ;;
+     -clang*)
+      arguments="$arguments /p:Compiler=$opt"
+      shift 1
+      ;;
+     -cmakeargs)
+      cmakeargs="${cmakeargs}x#160;${opt}x#160;$2"
```

Comment:
Why do we need the space encoded as `#160` here? Is the `/p:` unable to pass property with spaces even when it is enclosed in double quotes? Or do the double quotes get lost on their path through the various script levels?

Reply:
Problem is that we have too many indirections, and finally when msbuild is called from:

https://github.com/dotnet/runtime/blob/eb95163a0880cac1ba127b81e3cb753e5a65f8cc/eng/common/tools.sh#L396

it loses the quotes (single or double). If we echo this statement (until `||`) one line above and copy paste that line manually it works. But when it gets called from other script, it strips off the quotes and we get errors.

Reply:
> But when it gets called from other script, it strips off the quotes and we get errors.

which is strange and I do not know why it happens. (maybe echo statement expands `"$@"` differently, preserving the quote and actual msbuild command does not..).

Reply:
It is unfortunate we need to resort to working around it at several layers including the msbuild script. It seems it should be fixable in a clean way. I remember I was fighting with a similar thing when enabling Alpine support in core-setup repo long time ago and I think I was able to find a way to preserve the quotes. And I also remember echo was behaving differently w.r.t. "$@". I think that echo just always prints the double quotes. 
I'll try to look through my old changes to see if I can find it.

Reply:
It seems I was mistaken and I was doing something slightly different there. However, googling for a possible solution, I've realized that instead of #160, we might be able to use `\ `. Then we would not need to modify the msbuild scripts. Would you mind trying that?

Btw, in bash >= 4.4, we would be able to use ${@@Q} which quotes each parameter. But on my Ubuntu 16.04, I am still on 4.3, so it seems like 4.4 is too new to rely on.

Reply:
tracing with `bash -x` (in multiple places), turned out using `\ `, msbuild call from eng/common/tools.sh becomes:

> MSBuild /Users/adeel/.nuget/packages/microsoft.dotnet.arcade.sdk/5.0.0-beta.20071.3/tools/Build.proj /p:Configuration=Debug /p:RepoRoot=/Users/adeel/projects/runtime /p:Restore=true /p:Build=true /p:R
ebuild=false /p:Test=false /p:Pack=false /p:IntegrationTest=false /p:PerformanceTest=false /p:Sign=false /p:Publish=false /p:SubsetCategory=coreclr /p:Compiler=-clang /p:CrossBuild=True '/p:CMakeArgs="\' '-cm
akeargs\' '-DFeatureX=1\' '-cmakeargs\' 'Y=testing"'

vs. `x#160;` (xml encoding with & replaced by x):

> MSBuild /Users/adeel/.nuget/packages/microsoft.dotnet.arcade.sdk/5.0.0-beta.20071.3/tools/Build.proj /p:Configuration=Debug /p:RepoRoot=/Users/adeel/projects/runtime /p:Restore=true /p:Build=true /p:R
ebuild=false /p:Test=false /p:Pack=false /p:IntegrationTest=false /p:PerformanceTest=false /p:Sign=false /p:Publish=false /p:SubsetCategory=coreclr /p:Compiler=-clang /p:CrossBuild=True '/p:CMakeArgs="x#160;-
cmakeargsx#160;-DFeatureX=1x#160;-cmakeargsx#160;Y=testing"'

Reply:
I was misremembring which encoding scheme works natually with msbuild properties, it was URL-encoding instead of XML escaped values, which gets rendered in XML docs (`*proj` files). So replacing space with `%20` removed the need for manual replace.